core/src/main/java/ch/cyberduck/core/BookmarkCollection.java

```diff
@@ -91,6 +93,13 @@ public void collectionItemRemoved(final Host bookmark) {
                 log.debug("Skip indexing collection while loading");
             }
             else {
+                final Local file = this.getFile(bookmark);
+                try {
+                    LocalTrashFactory.get().trash(file);
```

Comment:
Still trashes session bookmarks during start in https://github.com/iterate-ch/cyberduck/blob/3fef1fb760e93d545d1f5c4da110ffb26eab13a0/osx/src/main/java/ch/cyberduck/ui/cocoa/controller/MainController.java#L956